### PR TITLE
feat(codegen,runtime): add two-argument Integer#pow(exp, mod)

### PR DESCRIPTION
**Pull request by Halvanhelv**
This adds the two-argument form of `Integer#pow(exp, mod)` for modular exponentiation. The one-argument form is unchanged.

The implementation uses binary exponentiation (repeated squaring) — the same algorithm CRuby uses via `rb_int_powm()`. Each intermediate result is taken `% mod` so values stay small regardless of exponent size.

Tests covering edge cases (zero base, mod=1, negative base, large exponent, base>mod, one-arg fallback) are in `test/integer_powmod.rb`.

### CODE REVIEW

File: lib/sp_runtime.h [hunk 1]
```diff
@@ -77,6 +77,7 @@ static inline mrb_int sp_imod(mrb_int a, mrb_int b) {
 
 static mrb_int sp_gcd(mrb_int a,mrb_int b){if(a<0)a=-a;if(b<0)b=-b;while(b){mrb_int t=b;b=a%b;a=t;}return a;}
 static mrb_int sp_lcm(mrb_int a,mrb_int b){if(a==0||b==0)return 0;mrb_int g=sp_gcd(a,b);if(a<0)a=-a;if(b<0)b=-b;return (a/g)*b;}
+static mrb_int sp_powmod(mrb_int base,mrb_int exp,mrb_int mod){mrb_int r=1;if(mod==1)return 0;base=base%mod;if(base<0)base+=mod;while(exp>0){if(exp%2==1)r=r*base%mod;exp=exp/2;base=base*base%mod;}return r;}
```
[the remark this answers is not in the document]
Comment (Halvanhelv) [reply lib/sp_runtime.h]: Medium (Negative Modulo):

Fixed.
